src/AssetManager/View/Helper/Asset.php

```diff
@@ -0,0 +1,140 @@
+<?php
+namespace AssetManager\View\Helper;
+
+use AssetManager\Asset\AggregateAsset;
+use AssetManager\Exception\InvalidArgumentException;
+use AssetManager\Service\AssetManager;
+use Zend\ServiceManager\ServiceLocatorInterface;
+use Zend\View\Helper\AbstractHelper;
+use Zend\Cache\Storage\Adapter\AbstractAdapter as AbstractCacheAdapter;
+
+class Asset extends AbstractHelper
+{
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $serviceLocator;
+
+    /**
+     * @var ServiceLocatorInterface
+     */
+    private $assetManager;
+
+    /**
+     * @param ServiceLocatorInterface $serviceLocator
+     * @param AssetManager            $assetManager
+     * @param array                   $config
+     */
+    public function __construct(ServiceLocatorInterface $serviceLocator, AssetManager $assetManager, $config)
+    {
+        $this->serviceLocator = $serviceLocator;
+        $this->assetManager   = $assetManager;
+        $this->config         = $config;
+    }
+
+    /**
+     * find the file and if it exists, append its unix modification time to the filename
+     *
+     * @param string $filename
+     * @param string $queryString
+     * @return string
+     */
+    private function elaborateFilePath($filename, $queryString)
+    {
+        /** @var AggregateAsset $asset */
+        $asset = $this->serviceLocator->get('AssetManager\Service\AssetManager')->getResolver()->resolve($filename);
+        if ($asset !== null) {
+
+            // append last modified date to the filepath and use a custom query string
+            return $filename . '?' . urlencode($queryString) . '=' . $asset->getLastModified();
+        }
+
+        return $filename;
+    }
+
+    /**
+     * Use the cache to get the filePath
+     *
+     * @param string $filename
+     * @param string $queryString
+     *
+     * @return mixed|string
+     */
+    private function getFilePathFromCache($filename, $queryString)
+    {
+        // check if the cache is configured
+        if (!isset($this->config['view_helper']['cache']) || $this->config['view_helper']['cache'] == null) {
+            return null;
+        }
+
+        // get the cache, if it's a string, search it among services
+        $cache = $this->config['view_helper']['cache'];
+        if (is_string($cache)) {
+            $cache = $this->serviceLocator->get($cache);
```

Comment:
https://github.com/RWOverdijk/AssetManager/pull/180/files#diff-25c0bbb35fb4899efece895d4b61b011R27

The factory does exactly that, inject dependencies. That's where you should get them :)

https://github.com/RWOverdijk/AssetManager/blob/master/src/AssetManager/Service/AssetFilterManagerServiceFactory.php#L27